File: 백준/Platinum/11003. 최솟값 찾기/최솟값 찾기.cc
#include <stdio.h>
#include <iostream>
#include <algorithm>
#include <deque>
#include <queue>

using namespace std;

int main()
{
	deque<int> dq;
	priority_queue<pair<int, int>, vector<pair<int, int>>, greater<>> q;

	int n, l, input;

	cin >> n >> l;
	for (int i = 0; i < n; i++) {
		scanf("%d", &input);

		dq.push_back(input);
		q.push({ input , i });

		auto t = q.top();
		while (q.top().second <= i - l) q.pop();
		printf("%d ", q.top());
	}
}

File: 백준/Platinum/11003. 최솟값 찾기/최솟값 찾기.py
from collections import deque
n, l = map(int, input().split())  
d = deque()
cur = list(map(int, input().split()))

for i in range(n):
  while d and d[-1][0] > cur[i]:
    d.pop()
  d.append((cur[i], i))
  if d[0][1] <= i - l:
    d.popleft()
  print(d[0][0], end=' ')